target/linux/ar71xx/files/arch/mips/ath79/mach-wnr2000-v5.c

```diff
@@ -0,0 +1,212 @@
+/*
+ *  NETGEAR WNR2000v5 board support
+ *
+ * Copyright (c) 2017 Raphael Catolino <raphael.catolino@gmail.com>
+ * Copyright (c) 2013-2015 The Linux Foundation. All rights reserved.
+ * Copyright (c) 2012 Gabor Juhos <juhosg@openwrt.org>
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#include <linux/gpio.h>
+#include <linux/platform_device.h>
+#include <linux/ath9k_platform.h>
+#include <linux/spi/74x164.h>
+#include <linux/spi/spi.h>
+#include <linux/spi/spi_gpio.h>
+
+#include <asm/mach-ath79/ath79.h>
+#include <asm/mach-ath79/ath79_spi_platform.h>
+#include <asm/mach-ath79/ar71xx_regs.h>
+
+#include "common.h"
+#include "dev-eth.h"
+#include "dev-gpio-buttons.h"
+#include "dev-leds-gpio.h"
+#include "dev-m25p80.h"
+#include "dev-spi.h"
+#include "dev-usb.h"
+#include "dev-wmac.h"
+#include "machtypes.h"
+
+#define WNR2000V5_GPIO_LED_STATUS_GREEN 21
+#define WNR2000V5_GPIO_LED_STATUS_AMBER 20
+#define WNR2000V5_GPIO_LED_DATA_GREEN   25
+#define WNR2000V5_GPIO_LED_DATA_AMBER   19
+#define WNR2000V5_GPIO_LED_WLAN         18
+
+#define WNR2000V5_GPIO_BTN_RFKILL 0
+#define WNR2000V5_GPIO_BTN_RESET  1
+#define WNR2000V5_GPIO_BTN_WPS    2
+
+#define WNR2000V5_KEYS_POLL_INTERVAL	20	/* msecs */
+#define WNR2000V5_KEYS_DEBOUNCE_INTERVAL	(4 * WNR2000V5_KEYS_POLL_INTERVAL)
+
+#define WNR2000V5_SPI_GPIO_RESET 14
+#define WNR2000V5_SPI_GPIO_MOSI 15
+#define WNR2000V5_SPI_GPIO_CLK 16
+#define WNR2000V5_74X164_GPIO_BASE QCA953X_GPIO_COUNT // The 74x614 muxer adds 8 new gpio after the regular 18 QCA953X gpios
```

Comment:
Wrong comment style, line too long, and it seems suspicious that the GPIO_BASE is the same as the GPIO_COUNT.

Reply:
Actually this is right. GPIO_COUNT corresponds to the number of gpios without the 74x164 muxer. That chip adds additional gpios and those are numbered starting from GPIO_COUNT.